usual/caslock.c

```diff
@@ -0,0 +1,55 @@
+#include <usual/logging.h>
+#include <usual/caslock.h>
+
+#define CAS_LOCK_UNLOCKED 0
+#define CAS_LOCK_LOCKED 1
+#define CAS_LOCK_INITIALIZED 1
+
+void cas_lock_init(CasLock *lock) {
```

Comment:
nit: Unless you have a good reason to prefer the name "CAS lock" I think "Spin lock" is generally better known name for this type of lock.